### DISCUSSION TITLE
How to watch the username and password field to reset the error message?

### DISCUSSION THREAD

**Discussion by web-programmer-here** [react-hook-form/react-hook-form#9541] [Q&A]
I am using below for onSubmit event for a form with two fields, username and password. 

```js
const [error, setError] = useState('')
const [success, setSuccess] = useState('')

const onSubmit = useCallback(
		async (data: FormData) => {
			try {
				await login(data)
				setSuccess('Successful login.')
			} catch (_) {
				setError('Failed login.')
			}
		},
		[login]
	)
```

At the top of the form, I want to display a message when user submits the form, an error when it fails and success message when user logged in successfully. 

```js
{success && {success} />}
{error && {error} />}
```

But I want to make the message disappear as the user types a new username or changes the password. Basically watch both fields and reset success or error. 

Based on this example: https://react-hook-form.com/api/usewatch

I created this: 

```js
function FirstNameWatched({ control }: { control: Control<FormData> }) {
		useWatch({
			control,
			name: ['username', 'password'],
		})

		setError('')
		setSuccess('')

		return (
			<>
				{success && {success} />}
                                {error && {error} />}
			</>
		)
	}
```

and called it within  my form

```js
<form ...>
<FirstNameWatched control={control} />
</form>
```

but now I cant see the message .

**Comment by bluebill1049** [comment]
could you try this: https://react-hook-form.com/api/useform/clearerrors?

**Comment by web-programmer-here** [comment]
@bluebill1049 thanks, this is not for individual errors, just one error when the user fails the login 

```js
await login(data)
setSuccess('Successful login.')
```

**Comment by Moshyfawn** [comment]
If your form is simple enough, you can make use of the form's `change` event to clear your login error:
```jsx
<form onChange={() => { if (error) setError("") }}>...</form>
```

The form's `change` event is fired when any of the nested form inputs are changed
